deploy/lib/util.rb

```diff
@@ -102,6 +102,36 @@ def file_exists?(filename)
   end
 end
 
+def get_files(path, options = {}, data = [])
+  ignore_extensions = ['..', '.', '.svn', '.git', '.ds_store', 'thumbs.db']
+
+  if File.directory?(path)
+    Dir.foreach(path) do |entry|
+      next if ignore_extensions.include?(entry.downcase)
+      full_path = File.join(path, entry)
+      skip = false
+
+      options[:ignore_list].each do |ignore|
+        if full_path.match(ignore)
+          skip = true
+          break
+        end
+      end if options[:ignore_list]
+
+      next if skip == true
+
+      if File.directory?(full_path)
+        get_files(full_path, options, data)
+      else
+        data << full_path.encode("UTF-8")
+      end
+    end
+  else
+    data = [path.encode("UTF-8")]
+  end
+  data
+end
```

Comment:
Actually, I'm using execute_query inside server_config to get the timestamps, so that code can't easily move inside xcc. In order to get the timestamps I need the list of files, so that also needs to be available to server_config. As I said, I could leave that in xcc, but it seemed more natural to put that in util with other filesystem utility methods.